### PR TITLE
Adding ML robustness analyses

### PR DESCRIPTION
**Pull request by gwaygenomics**
A series of notebooks performing three analyses:

1. Cell Line Holdout - Predict cell health readouts from one cell line completely held out of training
2. Sample Titration - Systematically remove samples and assess test set performance degradation
3. Feature Removal - Remove Cell Painting features and assess test set performance changes

### CODE REVIEW

**Comment by gwaygenomics** [comment]
## Cell line holdout analysis

cell_line_holdout_modz

**Figure legend. Predicting cell health readouts in held out cell lines.** We retrained a new model for each cell health variable using profiles from only 2 of 3 cell lines. We applied the models to predict the readouts from the third cell line. We show these predictions for each cell line when included in the held out set.

Even across cell lines and with a much smaller training set set (357 * 2/3 = 238), many models still detect signal more than a negative control. Many phenotypes can be predicted (e.g. viability, s phase, mitosis, death) in cell lines the model has never seen before, but many other phenotypes cannot be predicted. Not surprisingly, the shape phenotypes could not be predicted reliably in held out cell lines.

## Sample titration

sample_titration_modz

**Figure legend. The impact of removing samples from model training.** We show the mean test set R squared of three different sample titration iterations. (a) All 70 cell health models and the impact of removing samples prior to model training. (b) The mean test set R squared for all cell health targets. Note the difference in the y axis scale.

Removing all the way up to 8 samples (8/303 = 2.6%) performance actually increases in the test set (on average). However, once more samples are removed, the performance starts to decrease. Based on this analysis, it is difficult to conclude if adding samples actually would improve performance in general. I believe that I need more iterations of the sample dropping procedure, and more samples dropped, in order to better assess this question (ironic, I know!)

## Systematic feature group removal

feature_removal_modz

**Figure legend. Removing entire feature groups from training cell health models** We show the performance difference between the original cell health models and new models trained with the corresponding (a) feature group, (b) channel, and (c) compartment removed before model training. The `n` indicates the number of features removed and each point is an individual cell health model. (d) The impact of feature removal on individual cell health models.

As expected, systematically removing features does not have much of an impact on average for all feature classes. There are a handful of outliers, however. These outliers are driven mostly by a small minority of cell health models. It is also interesting that many of the models actually _increased_ in performance (negative values on x axis). For example, the nucleus roundess phenotype as predicted better when any set of features were removed.

**Comment by gwaygenomics** [comment]
TODO: I will increase the scope of the sample titration analysis - this takes a long time to run!

**Comment by gwaygenomics** [comment]
> TODO: I will increase the scope of the sample titration analysis - this takes a long time to run!

Done.
